code/modules/mob/grab/grab_datum.dm

```diff
@@ -167,37 +168,37 @@
 
 
 /datum/grab/proc/adjust_position(var/obj/item/grab/G)
-	var/mob/living/carbon/human/affecting = G.affecting
-	var/mob/living/carbon/human/assailant = G.assailant
+
+	var/atom/movable/affecting = G.affecting
+	var/mob/living/assailant = G.assailant
 	var/adir = get_dir(assailant, affecting)
 
 	if(same_tile)
 		affecting.forceMove(assailant.loc)
 		adir = assailant.dir
 		affecting.set_dir(assailant.dir)
 
-	switch(adir)
-		if(NORTH)
-			animate(affecting, pixel_x = 0, pixel_y =-shift, 5, 1, LINEAR_EASING)
-			G.draw_affecting_under()
-		if(SOUTH)
-			animate(affecting, pixel_x = 0, pixel_y = shift, 5, 1, LINEAR_EASING)
-			G.draw_affecting_over()
-		if(WEST)
-			animate(affecting, pixel_x = shift, pixel_y = 0, 5, 1, LINEAR_EASING)
-			G.draw_affecting_under()
-		if(EAST)
-			animate(affecting, pixel_x =-shift, pixel_y = 0, 5, 1, LINEAR_EASING)
-			G.draw_affecting_under()
-
-	affecting.reset_plane_and_layer()
+	if(ismob(affecting) && !istype(affecting, /mob/living/exosuit)) // Structures often have mapped pixel offsets - donut touch.
```

Comment:
Make it a mob (or movable) proc or flag or something? Maybe even put the logic there, not just the check.

Reply:
Broke the logic out to procs on the relevant subtypes, seemed cleaner.